### ISSUE TITLE
[NUCLEO_F091RC] ADC reading of internal sensors 

### ISSUE THREAD

**Issue by rolandvs** [micropython/micropython#3917]
While trying the pyb.ADCAll() function I came across a possible bug:
Given:
```
import pyb
adc=pyb.ADCAll(12, 0x70000)
print(adc.read_core_temp())
```
Will print the temperature of the chip around 36.35 degrees while writing this. Whenever another core function is read, reading the temperature again will show false readings like 73.358 degrees.
Using a soft reset (ctrl-D) will not help to get the first situation back, a hard reset is necessary.

**Comment by dpgeorge** [comment]
Yes I did notice the temp reading was not working in some cases on this board.  I don't know the exact reason, it needs some investigation.

**Comment by rolandvs** [comment]
Have tested It again and it seems that the first time it is working, however if you have used `read_core_vref()` subsequent readings will give a false value only to cleared when a Power On Reset is given (ctrl-D will not work).

**Comment by dpgeorge** [comment]
This should be fixed by 7be5bb367212b6949e74f73d90af01f8b68f1352

**Comment by rolandvs** [comment]
Verified and working!
